### PR TITLE
Benchmark

### LINKED ISSUE #2: Benchmark
Similar to https://github.com/sindresorhus/strip-css-comments/blob/master/bench.js

It should already be pretty fast, but you never really know until you bench it. I'm sure there are ways to further increase the performance.

Help appreciated :)

### PR DESCRIPTION
**Pull request by daveseco7**
The requested feature is implemented in the `bench.js` file. As it stands it, this benchmark makes use of a gigantic text source (`fixture.txt`), thus providing a better view of the implemented solutions. To better analyse each pattern execution, each pattern is benchmarked individually, as well as in conjunction with other patterns. 
A small linter error was fixed in the `index.js` file, as it was breaking the tests execution.
Currently, matcher relies on multiple modules that require **node version >= 4**, for that reason node engine was upped to >=4 and Travis **node versions 0.10 and 0.12 were removed**.
Added matcha as dev dependency and support for node engine 6 and 7.

---

Fixes #2

### CODE REVIEW

**Comment by sindresorhus** [comment]
Looks great. Can you add some `bench`'s for one paragraph and for one sentence? Just to see how it fares for shorter strings. In AVA, we use it mostly for (short) test titles, so interesting to see the performance for that too.

File: bench.js [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+bench('matcher-test-negation', function () {
+	matcher(fixture.split(' '), ['!foo']);
+});
+
+bench('matcher-test-negation-zero-or-more', function () {
+	matcher(fixture.split(' '), ['!*foo']);
+});
+
+bench('is-match-test', function () {
```
Comment (sindresorhus) [inline bench.js]: Can you drop the `-` in the titles?

File: bench.js [hunk 2]
```diff
@@ -0,0 +1,26 @@
+'use strict';
+/* global bench */
+var fs = require('fs');
+var matcher = require('./');
+
+var fixture = fs.readFileSync('fixture.txt', 'utf8');
```
Comment (sindresorhus) [inline bench.js]: Use `const` for these.

File: bench.js [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+
+bench('matcher-test-multiple-patterns', function () {
```
Comment (sindresorhus) [inline bench.js]: And arrow functions.

File: bench.js [hunk 4]
```diff
@@ -0,0 +1,26 @@
+'use strict';
+/* global bench */
+var fs = require('fs');
+var matcher = require('./');
```
Comment (sindresorhus) [inline bench.js]: In Node.js 4, you can actually simplify `./` to `.`

**Comment by daveseco7** [comment]
I may have been a little overboard with the number of benchmark methods, but this way we can extrapolate more precise results, regarding the set of data being tested.

File: bench.js [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+bench('paragraph matcher test zero or more chars', () => matcher(paragraph.split(' '), ['*foo']));
+bench('paragraph matcher test negation', () => matcher(paragraph.split(' '), ['!foo']));
+bench('paragraph matcher test negation zero or more', () => matcher(paragraph.split(' '), ['!*foo']));
+bench('paragraph is match test', () => matcher.isMatch(paragraph, '*bar*'));
+
+bench('fixture matcher test multiple patterns', () => matcher(fixture.split(' '), ['*bar', '!foo', '!*oo']));
+bench('fixture matcher test zero or more chars', () => matcher(fixture.split(' '), ['*foo']));
+bench('fixture matcher test negation', () => matcher(fixture.split(' '), ['!foo']));
+bench('fixture matcher test negation zero or more', () => matcher(fixture.split(' '), ['!*foo']));
+bench('fixture is match test', () => matcher.isMatch(fixture, '*bar*'));
```
Comment (sindresorhus) [inline bench.js]: I think `isMatch` and `matcher` should be split up into separate `suite`'s to improve the output.

Comment (daveseco7) [inline bench.js]: I do agree with you that a user may be overwhelmed with the output and may not distinguish between `isMatch` and `matcher` benchmark results. I'm having a bit of trouble in understanding what do you mean with `separate suite's`. Could you please clarify me?

Comment (sindresorhus) [inline bench.js]: Matcha supports grouping benchmarks in `suit`'s, like Mocha:

```
suite('Make Tea', function () {
```

https://github.com/logicalparadox/matcha

**Comment by sindresorhus** [comment]
Awesome. Thanks you @daveseco7 :)

**Pull request merged**